design-library/src/components/BccProgress/BccProgress.vue

```diff
@@ -0,0 +1,74 @@
+<script lang="ts">
+export default {
+  inheritAttrs: false,
+};
+</script>
+
+<script setup lang="ts">
+import { computed, type StyleValue } from "vue";
+import BccFormLabel from "@/components/BccFormLabel/BccFormLabel.vue";
+
+type Props = {
+  value?: number;
+  max?: number;
+  size?: "sm" | "base" | "lg";
+  showValues?: boolean;
+  showPercentage?: boolean;
+  exceed?: boolean;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  value: 0,
+  max: 100,
+  size: "base",
+  showValues: true,
+  showPercentage: true,
+  exceed: true,
+});
+
+const percentage = computed(() => {
+  if (props.value > props.max && !props.exceed) return 100;
+  return (props.value / props.max) * 100;
+});
+
+const percentageSafe = computed(() => {
+  if (props.value > props.max) return 100;
+  return percentage.value;
+});
+
+const height = computed(() => {
+  return {
+    "h-1": props.size === "sm",
+    "h-1.5": props.size === "base",
+    "h-2": props.size === "lg",
+  };
+});
+</script>
+
+<template>
+  <div
+    class="bcc-progress-container"
+    :class="$attrs['class']"
+    :style="$attrs['style'] as StyleValue"
```

Comment:
Vue will add any `class` and `style` elements by default on the outer element, so if that's the intention then this is not needed. In some cases, like the Input component, it seemed to make more sense to add classes to the `input` element itself, not the wrapping `div`. But in this case I think it's fine to put the classes on here, and then this code is not needed.